packages/web/src/lib/myalgowallet-mode.ts

```diff
@@ -146,25 +165,47 @@ export class MyAlgoWalletSession {
 	async executeTx(
 		execParams: ExecParams[]
 	): Promise<algosdk.modelsv2.PendingTransactionResponse> {
-		let signedTxn;
+		let signedTxn: SignedTx[] | undefined;
 		let txns: Transaction[] = [];
 		if (execParams.length > 16) {
 			throw new Error("Maximum size of an atomic transfer group is 16");
 		}
+
+		if (isSDKTransactionAndSign(execParams[0]))
+			throw new Error("We don't support this case now");
+
 		for (const [_, txn] of execParams.entries()) {
 			txns.push(mkTransaction(txn, await mkTxParams(this.algodClient, txn.payFlags)));
 		}
 
 		txns = algosdk.assignGroupID(txns);
-		const toBeSignedTxns: TransactionInGroup[] = txns.map((txn: Transaction) => {
-			return { txn: txn, shouldSign: true };
-		});
 
-		signedTxn = await this.signTransactionGroup(toBeSignedTxns);
+		// with logic signature we set shouldSign to false
+		const toBeSignedTxns: TransactionInGroup[] = execParams.map(
+			(txn: ExecParams, index: number) => {
+				return txn.sign === SignType.LogicSignature
```

Comment:
Use switch case here. Check executeTx of AlgoSigner.

Reply:
why do we need switch case, if only single if is there, switch makes more sense when more than 2/3 cases are ther

Reply:
We have 3 SignType now. Currently, in this method we are not using but in case if we decide to added in future then it will be small change.

Reply:
done